kafka-rest/src/main/java/io/confluent/kafkarest/response/CrnFactoryImpl.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.response;
+
+import io.confluent.kafkarest.config.ConfigModule.CrnAuthorityConfig;
+import java.util.StringJoiner;
+import javax.inject.Inject;
+
+/**
+ * A {@link CrnFactory} that uses the given {@code crnAuthorityConfig} as the Confluent Resource
+ * Authority for the generated CRNs.
+ */
+public final class CrnFactoryImpl implements CrnFactory {
+
+  private static final char SEPARATOR = '/';
+
+  private final String baseCrn;
+
+  @Inject
+  public CrnFactoryImpl(@CrnAuthorityConfig String crnAuthorityConfig) {
```

Comment:
not sure i like DI to this level of granularity. but minor point, and i'm not sure of this opinion.